diff --git a/node_modules/ink-text-input/build/index.js b/node_modules/ink-text-input/build/index.js
index ada7cdc..2288700 100644
--- a/node_modules/ink-text-input/build/index.js
+++ b/node_modules/ink-text-input/build/index.js
@@ -90,11 +90,11 @@ function TextInput({ value: originalValue, placeholder = '', focus = true, mask,
                 nextCursorWidth = input.length;
             }
         }
-        if (cursorOffset < 0) {
+        if (nextCursorOffset < 0) {
             nextCursorOffset = 0;
         }
-        if (cursorOffset > originalValue.length) {
-            nextCursorOffset = originalValue.length;
+        if (nextCursorOffset > nextValue.length) {
+            nextCursorOffset = nextValue.length;
         }
         setState({
             cursorOffset: nextCursorOffset,
